2-Draft-Accepted/RFCNNNN-Support-Experimental-Features.md

```diff
@@ -308,6 +316,8 @@ As for the type members with the `[Experimental()]` attributes,
 [`TypeDefiner`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/PSType.cs#L17) needs to be updated to process or skip a type member when processing a `TypeDefinitionAst` node,
 depending on whether the experimental feature is enabled and the `FeatureAction` value.
 
+> NOTE: When are errors generated, if any, if an experimental script function or class member is referenced, such as within a script module or within the class implementation?
```

Comment:
The errors will be generated at runtime. For a type reference, powershell will throw "type cannot be found". For a member reference, it will throw "member cannot be found".

Reply:
For script functions, are the functions stripped or are stubs used to throw an explicit error.  My thought we could end up with unintended false positives resolving to something else.

Reply:
For script functions, if the associated experimental feature is not enabled, the functions will not be logged in the function table, so any reference to the script function would result in a "function not found" error at runtime.

For classes, if a class is referencing another class that is experimental, then depending on the manner of the reference, the former class itself or members of it should be marked with the same experimental feature.
Scenarios:
1. `class Foo : Bar {...}` -- `Bar` is an experimental feature, then `Foo` must also be an experimental feature.
2. `class Foo { [Bar] $member }` -- `Bar` is an experimental feature, then `$member` must also be an experimental feature.